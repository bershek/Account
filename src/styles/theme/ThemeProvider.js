import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import { selectTheme } from './slice/selectors';
export var ThemeProvider = function (props) {
    useThemeSlice();
    var theme = useSelector(selectTheme);
    return (React.createElement(OriginalThemeProvider, { theme: theme }, React.Children.only(props.children)));
};
