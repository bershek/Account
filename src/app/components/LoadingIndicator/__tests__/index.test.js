var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { render } from '@testing-library/react';
import { LoadingIndicator } from '../index';
import { themes } from 'styles/theme/themes';
import { ThemeProvider } from 'styled-components';
var renderWithTheme = function (props, theme) {
    if (props === void 0) { props = {}; }
    return render(React.createElement(ThemeProvider, { theme: theme || themes.light },
        React.createElement(LoadingIndicator, __assign({}, props))));
};
describe('<LoadingIndicator />', function () {
    it('should match snapshot', function () {
        var loadingIndicator = renderWithTheme();
        expect(loadingIndicator.container.firstChild).toMatchSnapshot();
    });
    it('should match snapshot when props changed', function () {
        var loadingIndicator = renderWithTheme({ small: true });
        expect(loadingIndicator.container.firstChild).toMatchSnapshot();
    });
    it('should have theme', function () {
        var loadingIndicator = renderWithTheme();
        expect(loadingIndicator.container.querySelector('circle')).toHaveStyle("stroke: ".concat(themes.light.primary));
    });
});
