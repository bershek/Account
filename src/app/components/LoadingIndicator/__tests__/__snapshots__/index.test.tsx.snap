// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should match snapshot 1`] = `
<svg
  class="LoadingIndicator__Svg-sc-1uoud3c-0 eSJumQ"
  viewBox="-24 -24 48 48"
>
  <circle
    class="LoadingIndicator__Circle-sc-1uoud3c-1 jnnFam"
    cx="0"
    cy="0"
    fill="none"
    r="20"
    stroke-width="4"
  />
</svg>
`;

exports[`<LoadingIndicator /> should match snapshot when props changed 1`] = `
<svg
  class="LoadingIndicator__Svg-sc-1uoud3c-0 lgggPg"
  viewBox="-24 -24 48 48"
>
  <circle
    class="LoadingIndicator__Circle-sc-1uoud3c-1 jnnFam"
    cx="0"
    cy="0"
    fill="none"
    r="20"
    stroke-width="4"
  />
</svg>
`;
